contract/contracts/Subscriptions.sol

```diff
@@ -99,40 +87,53 @@ contract Subscriptions {
     // and cannot be recovered by the user.
     // Defined as `rate * max(0, min(block, end) - start)`
     function locked(Subscription storage sub) private view returns (uint128) {
-        uint64 currentBlock = uint64(block.number);
-        int256 len = Prelude.max(
-            0,
-            Prelude.min(int64(currentBlock), int64(sub.end)) - int64(sub.start)
+        uint256 len = uint256(
+            SignedMath.max(
+                0,
+                int256(Math.min(block.number, sub.end)) - int64(sub.start)
+            )
         );
-        return sub.rate * uint128(uint256(len));
+        return sub.rate * uint128(len);
     }
 
     // Unlocked tokens for a subscription are not collectable by the contract
     // owner and can be recovered by the user.
     // Defined as `rate * max(0, end - max(block, start))`
-    function unlocked(Subscription storage sub) private view returns (uint128) {
-        uint64 currentBlock = uint64(block.number);
-        int256 len = Prelude.max(
-            0,
-            int64(sub.end) - Prelude.max(int64(currentBlock), int64(sub.start))
+    function unlocked(Subscription memory sub) private view returns (uint128) {
+        uint256 len = uint256(
+            SignedMath.max(
+                0,
+                int256(int64(sub.end)) -
+                    int256(Math.max(block.number, sub.start))
+            )
         );
```

Comment:
@tmigone it would appear that solidity disagrees with you
![Screenshot 2023-02-02 at 8 43 09 AM](https://user-images.githubusercontent.com/18075124/216420194-13c568ef-0e79-40ff-9044-5ba3aa9a4091.png)

Reply:
ahh forgot `sub.end` was `uint`.